documentation/creating_a_script_plan.md

```diff
@@ -0,0 +1,369 @@
+  docker-compose pull $c
+done
+docker-compose up -d --remove-orphans
+```
+
+To pass the argument to the script as part of your `manage_docker::update_images` plan, [add a
+plan parameter](writing_yaml_plans.md#parameters) for the argument and pass it through to
+the script. In a YAML plan, that should look something like this:
+```yaml
+parameters:
+  directory:
+    type: Optional[String]
+    default: .
+    description: The directory to execute 'docker-compose' from
+  targets:
+    ...
+```
+
+Next, add the argument to the [script step](writing_yaml_plans.md#script_step)
+```yaml
+steps:
+  - name: run_script
+    script: manage_docker/scripts/update_images.sh
+    arguments:
+      - $directory
+    targets: $targets
+```
+
+And now your plan should look something like this:
+```yaml
+description: Update Docker images
+parameters:
+  directory:
+    type: Optional[String]
+    default: .
+    description: The directory to execute 'docker-compose' from
+  targets:
+    type: TargetSpec
+    description: A list of targets to run actions on
+
+steps:
+  - name: run_script
+    script: manage_docker/scripts/update_images.sh
+    arguments:
+      - $directory
+    targets: $targets
+
+return: $run_script
+```
+
+You can run the plan with:
+
+_\*nix shell command_
+```
+bolt plan run manage_docker::update_images -t <targets> directory=./docker
+```
+
+_PowerShell cmdlet_
+```powershell
+Invoke-BoltPlan -Name manage_docker::update_images -Targets <targets> directory=./docker
+```
+
+### As PowerShell parameters
+
+PowerShell parameters are a more fine-grained way to specify inputs for PowerShell scripts than
+Bash-like command-line arguments. Using PowerShell parameters is highly recommended to more easily
+pass arguments on the command-line and through plans to PowerShell scripts.
+
+For example, if you have a script to reboot your machine that takes a `timeout` parameter to specify how
+long to wait for the reboot, and a `shutdown_only` parameter to tell the script not to turn the
+machine back on:
+
+_reboot.ps1_
+```
+[CmdletBinding()]
+Param(
+  [Int]$timeout = 3,
+  [Boolean]$shutdown_only = $false
+)
+If (Test-Path -Path $env:SYSTEMROOT\sysnative\shutdown.exe) {
+  $executable = "$env:SYSTEMROOT\sysnative\shutdown.exe"
+}
+ElseIf (Test-Path -Path $env:SYSTEMROOT\system32\shutdown.exe) {
+  $executable = "$env:SYSTEMROOT\system32\shutdown.exe"
+}
+Else {
+  $executable = "shutdown.exe"
+}
+
+# Force a minimum timeout of 3 second to allow the response to be returned.
+If ($timeout -lt 3) {
+  $timeout = 3
+}
+
+$reboot_param = "/r"
+If ($shutdown_only) {
+  $reboot_param = "/s"
+}
+
+& $executable $reboot_param /t $timeout /d p:4:1
+```
+
+After turning the script into a plan with `bolt plan new --script` or `New-BoltPlan -Script`, add
+the parameters for your script to the plan:
+```yaml
+parameters:
+  timeout:
+    type: Optional[Integer]
+    default: 3
+    description: How long to wait for reboot before exiting
+  shutdown:
```

Comment:
```suggestion
  shutdown_only:
```

The plan and `plan run` command both use `shutdown`, but the script uses `shutdown_only`. One or the other should be changed to be consistent.